docs/stdlib/cfg.rst

```diff
@@ -0,0 +1,148 @@
+.. _ref_std_cfg:
+
+======
+Config
+======
+
+The ``cfg`` module contains a set of types and scalars used for configuring
+EdgeDB.
+
+
+.. list-table::
+  :class: funcoptable
+
+  * - **Type**
+    - **Description**
+  * - :eql:type:`cfg::AbstractConfig`
+    - The base type for all configuration objects. The properties of this type
+      define the set of configuruation settings supported by EdgeDB.
+  * - :eql:type:`cfg::Auth`
+    - An object type representing an authentication profile.
+  * - :eql:type:`cfg::AuthMethod`
+    - An abstract object type representing a method of authentication
+  * - :eql:type:`cfg::Trust`
+    - A subclass of ``AuthMethod`` indicating an "always trust" policy (no
+      authentication).
+  * - :eql:type:`cfg::SCRAM`
+    - A subclass of ``AuthMethod`` indicating password-based authentication.
+
+
+----------
+
+
+.. eql:type:: cfg::AbstractConfig
+
+  An abstract type representing the configuration of an instance or database.
+
+  Is extended by three non-abstract subclasses:
+  ``cfg::Config``, ``cfg::InstanceConfig``, and ``cfg::DatabaseConfig``. At
+  the moment, these subclasses do not declare any additional properties or
+  links beyond what they inherit from ``cfg::AbstractConfig``.
+
+  The properties of this object type represent the set of configuration
+  options supported by EdgeDB.
+
+
+  .. list-table::
+
+    * - **Setting**
+      - **Type**
+      - **Default**
+    * - ``client_idle_timeout``
+      - ``required std::int16``
+      - ``30 seconds``
+    * - ``listen_port``
+      - ``required std::int16``
+      - ``5656``
+    * - ``listen_addresses``
+      - ``multi std::str``
+      - N/A
+    * - ``auth``
+      - ``multi cfg::Auth``
+      - N/A
+    * - ``allow_dml_in_functions``
+      - ``std::bool``
+      - N/A
+    * - ``shared_buffers``
```

Comment:
Can we add copy from guides/configuration to here?

Reply:
I think that should probably stay as a guide (though it needs some work)

Reply:
Added a setting reference to this page